### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by foxpy** [foxpy/dmotd#4]
- [ ] Public IP
- [ ] Network total RX/TX stats
- [ ] Config file

**Comment by foxcpp** [comment]
It might be a good idea to cache Public IP information since it might be slow to obtain and this may cause shell logins to be slow if the server is under heavy load.

**Comment by foxpy** [comment]
But how should it act in case if public IP has changed? Just spit out cached IP, fork into background and update it? Or may it be just enough to use some small timeout?

**Comment by foxcpp** [comment]
Yeah, timeout would be a good idea. It is hard to pick a reasonable value, something like 1 hour might work. It is important to make it configurable though and preferably update it using cron/systemd.

**Comment by foxpy** [comment]
1 hour? Sounds like too much time wasted by user logging into console.
The best scheme should be:

1) User logs in, triggering the launch of dmotd;
2) dmotd checks for external IP address with a very little timeout (possibly 1 second);
3.1) On success, the cached value is updated and printed with a (fresh) timestep of cached value.
3.2) On failure, the cached value is printed with a (possibly pretty old) timestep of cached value.

In addition to this, systemd/cron timer updates cached value from time-to-time.
